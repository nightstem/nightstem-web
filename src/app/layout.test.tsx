import { axe } from 'vitest-axe';
import { describe, expect, it, vi } from 'vitest';
import { render, screen, act } from '@testing-library/react';

import Layout, { metadata, viewport } from './layout';
import type { Metadata, Viewport } from 'next';

vi.mock('next/font/google', () => {
  const actual = vi.importActual('next/font/google');
  return {
    ...actual,
    Geist: vi.fn().mockReturnValue({ variable: 'geist' }),
    Geist_Mono: vi.fn().mockReturnValue({ variable: 'geist_mono' }),
  };
});

vi.mock('@growthbook/growthbook-react', () => ({
  GrowthBook: vi.fn().mockImplementation(() => ({
    init: vi.fn().mockResolvedValue(undefined),
    destroy: vi.fn(),
    isReady: vi.fn().mockReturnValue(true),
    ready: vi.fn().mockResolvedValue(undefined),
    getFeatureValue: vi.fn().mockReturnValue(false),
  })),
  GrowthBookProvider: ({ children }: { children: React.ReactNode }) => children,
  useFeatureValue: vi.fn().mockReturnValue(false),
}));

describe('metadata', () => {
  it('has the correct information', () => {
    expect(metadata).toEqual({
      title: 'Nightstem',
      description:
        'Nightstem is a software studio creating thoughtful, minimal tools and digital experiences â€” quietly crafted after hours.',
    } satisfies Metadata);
  });
});

describe('viewport', () => {
  it('has the correct configuration', () => {
    expect(viewport).toEqual({
      width: 'device-width',
      initialScale: 1,
    } satisfies Viewport);
  });
});

describe('Layout', () => {
  const childrenText = 'Default Children Text';

  it('matches snapshot', async () => {
    let container;
    await act(async () => {
      const rendered = render(<Layout>{childrenText}</Layout>);
      container = rendered.container;
    });
    expect(container).toMatchSnapshot();
  });

  it('renders children correctly', async () => {
    await act(async () => render(<Layout>{childrenText}</Layout>));

    expect(await screen.findByText(childrenText)).toBeInTheDocument();
  });

  it('does not have accessibility violations', async () => {
    let container: Element;
    await act(async () => {
      const rendered = render(<Layout>{childrenText}</Layout>);
      container = rendered.container;
    });
    const result = await axe(container!);
    expect(result).toHaveNoViolations();
  });
});
