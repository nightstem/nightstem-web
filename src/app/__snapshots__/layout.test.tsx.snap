// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col h-dvh w-dvw"
  >
    <nav
      class="w-full flex justify-center"
    >
      <div
        class="screen-container px-2 py-2.5 flex items-center justify-between"
      >
        <div
          class="inline-flex items-center gap-2"
        >
          <svg
            aria-hidden="true"
            class="text-primary-500 h-[1.066rem] w-[1.066rem] motion-reduce:animate-none shrink-0"
            fill="none"
            viewBox="0 0 500 500"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M346.04 135.875L500 245.427L161.185 495.252C157.629 497.874 151.828 497.923 148.228 495.362L0.786259 390.448L346.04 135.875Z"
              fill="currentColor"
            />
            <path
              d="M148.322 4.74771C151.878 2.12562 157.68 2.07657 161.279 4.63814L333.785 127.386L179.023 241.499L9.54552e-05 114.113L148.322 4.74771Z"
              fill="currentColor"
            />
          </svg>
          <span
            class="heading-sm"
          >
            Nightstem
          </span>
        </div>
        <ol>
          <li>
            <a
              class="select-none whitespace-nowrap font-medium cursor-pointer disabled:cursor-not-allowed transition-colors duration-200 motion-reduce:transition-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 ring-offset-dark text-sm px-3 py-1.5 rounded-md text-neutral-400 hover:text-neutral-300 active:text-neutral-300 focus-visible:ring-neutral-400 disabled:opacity-60 disabled:hover:text-neutral-400 disabled:hover:bg-transparent disabled:active:bg-transparent"
              href="/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
    </nav>
    <main
      class="screen-container h-full max-h-full overflow-auto"
    >
      Default Children Text
    </main>
  </div>
</div>
`;
